### ISSUE TITLE
Violation of `simple-unless` allowlist doesn't get added to TODOs when using `--update-todo`

### ISSUE THREAD

**Issue by cincodenada** [ember-template-lint/ember-template-lint#2042]
## Summary

I've created a minimal test case that is a freshly-generated ember application with two small commits demonstrating the issue [here](https://github.com/cincodenada/etl-bug-example). I'll summarize the important bits here:

I have the following `application.hbs`:
```hbs
{{#unless (or true false)}}
This should be fine
{{/unless}}

{{#unless (or (or true false) false)}}
This should not
{{/unless}}

{{#unless (or (and true false) false)}}
This is a mess
{{/unless}}

{{outlet}}
```

And this `.template-lintrc.js`:
```js
'use strict';

module.exports = {
  extends: 'recommended',
  rules: {
    'simple-unless': {
      allowlist: ['or'],
      maxHelpers: 1
    }
  }
};
```

When I run linting, the first `unless` block passes and the next two don't, as expected. The second block has two separate violations:
```shell
 in etl-bug/ on before-todo
› npx ember-template-lint .
app/templates/application.hbs
  5:14  error  Using {{unless}} in combination with other helpers should be avoided. MaxHelpers: 1  simple-unless
  9:14  error  Using {{unless}} in combination with other helpers should be avoided. MaxHelpers: 1  simple-unless
  9:14  error  Using {{unless}} in combination with other helpers should be avoided. Allowed helper: or  simple-unless

✖ 3 problems (3 errors, 0 warnings)
```

When I try to mark it as TODO, only the `MaxHelpers` violations get added. the `Allowed helper` violation stubbornly remains as an active violation, and I can't figure out how to mark it as TODO:
``` shell
 in etl-bug/ on before-todo
› npx ember-template-lint . --update-todo
app/templates/application.hbs
  9:14  error  Using {{unless}} in combination with other helpers should be avoided. Allowed helper: or  simple-unless

✖ 1 problems (1 errors, 0 warnings)
✔ 2 todos created, 0 todos removed (warn after 30, error after 60 days)

 in etl-bug/ on before-todo
› npx ember-template-lint .
app/templates/application.hbs
  9:14  error  Using {{unless}} in combination with other helpers should be avoided. Allowed helper: or  simple-unless

✖ 1 problems (1 errors, 0 warnings)
```

## Expected Behavior:
All three violations get added marked as TODOs and no longer show up in the regular `ember-template-lint` output
## Actual behavior:
The `Allowed helper` violation doesn't get added as a TODO and continues to show up as a violation after running `--update-todo`

Happy to provide any other info - the [minimal example linked above](https://github.com/cincodenada/etl-bug-example) should be in the broken state where the first two are added as TODOs but the second isn't, and refuses to be added. You can check out the `before-todo` branch (or just remove `.lint-todo`) to see things before the TODOs are added.

**Comment by lifeart** [comment]
Moving issue to umbrella ticket https://github.com/ember-template-lint/ember-template-lint/issues/new and closing it here (for now)
